nidata/multimodal/hcp/__init__.py

```diff
@@ -89,11 +89,42 @@ def prepend(self, src_files):
                 files.append((src_file, 'HCP/' + src_file))
         return files
 
-    def get_subject_list(self, n_subjects=500):
+    def get_subject_list(self, n_subjects=None):
         """Get the list of subject IDs. Depends on the # of subjects,
         which also corresponds to other things (license agreement,
         type of data available, etc)"""
-        return ['100307']  # 992774']
+        subjectfile = 'S900.txt'
+        while True:
+            try:
+                fil = self.fetcher.fetch(self.prepend([subjectfile]))[0]
+                break
+            except:
+                print 'Error while fetching file S900.txt. Using file S500.txt instead.'
+                subjectfile = 'S500.txt'
+                try:
+                    fil = self.fetcher.fetch(self.prepend([subjectfile]))[0]
+                    break
+                except:
+                    print 'Error while fetching file S500.txt. Using file UR100.txt instead.'
+                    subjectfile = 'UR100.txt'
+                    try:
+                        file = self.fetcher.fetch(self.prepend([subjectfile]))[0]
+                        break
+                    except:
+                        print 'Error while fetching file UR100.txt. Dataset fetching aborted.'
+        if subjectfile=='S900.txt' and n_subjects>900:
+            raise IndexError('Subjects number requested too high.' \
+                  'Please enter a number less than or equal to 900.')
+        if subjectfile=='S500.txt' and n_subjects>500:
+            raise IndexError('Subjects number requested too high.' \
+                  'Please enter a number less than or equal to 500.')
+        if subjectfile=='UR100.txt' and n_subjects>100:
+            raise IndexError('Subjects number requested too high.' \
+                  'Please enter a number less than or equal to 100.')
```

Comment:
@hwd15508 I think this code will work, but there's a lot of code duplication. In the `while` loop above, there's no actual loop--except an infinite one if all files fail to download.

Try a loop something like this:
```python
subj_file_info = (('S900.txt', 900), ('S500.txt', 500), ('U100.txt', 100))
fil = None
for fname, nsubj in subj_file_info:
    try:
        fil = self.fetcher.fetch(self.prepend([fname]))
        break
    except:
        print...

# Completed the loop. Make sure we succeeded, and that what's requested is possible. 
if fil is None:
    raise Exception('Failed to fetch subject list from any...')
if n_subjects > nsubj:
    raise IndexError(...)
```